File: Core/Entities/PaginatedResult.cs
namespace Core.Entities
{
    public class PaginatedResult<T>
    {
        public List<T> Data { get; set; }
        public int PageSize { get; set; }
        public int PageNum { get; set; }
        public int TotalCount { get; set; }
        public int TotalPages => (int)Math.Ceiling((double)TotalCount / PageSize);
    }
}

File: client/src/app/shared/models/PaginatedResult.ts
export class PaginatedResult<T> {
    data: T[];
    pageSize: number;
    pageNum: number;
    totalCount: number;
    totalPages: number;
  
    constructor(data: T[], pageSize: number, pageNum: number, totalCount: number) {
      this.data = data;
      this.pageSize = pageSize;
      this.pageNum = pageNum;
      this.totalCount = totalCount;
      this.totalPages = Math.ceil(totalCount / pageSize);
    }
  }